### PR TITLE
feat(feature-manager): remove Intranet from feature manager

### PR DESCRIPTION
**Pull request by mantariksh**
## Problem

Part of #1842 

## Solution

1. Remove the Intranet feature from the feature manager.
2. Remove the IntranetFactory and link the service directly to its consumers. The IntranetFactory previously served simply to wrap the service in a `Result` in case the feature was not activated, so it is no longer necessary.
3. Allow the intranet IP list path to default to an empty string, which means that no IPs are detected as intranet IPs.

**Breaking Changes** 

- [ ] Yes - this PR contains breaking changes
    - Details ...
- [x] No - this PR is backwards compatible  
    - Server-side only

## Deploy Notes

**Note that the server will now crash in the absence of the `INTRANET_IP_LIST_PATH` environment variable.**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -39,6 +39,7 @@ services:
       - MYINFO_CLIENT_ID=mockClientId
       - MYINFO_CLIENT_SECRET=mockClientSecret
       - WEBHOOK_SQS_URL=http://localhost:4566/000000000000/local-webhooks-sqs-main
+      - INTRANET_IP_LIST_PATH=tests/mock-intranet-ips.txt
```
Comment (karrui) [inline docker-compose.yml:42]: does this need to be added to docker?

**Review by karrui — APPROVED** [review]
lgtm with question

File: docker-compose.yml (hunk 1 above)
Comment (mantariksh) [reply docker-compose.yml:42]: ? this is the docker-compose file

Comment (mantariksh) [reply docker-compose.yml]: as discussed, see https://github.com/opengovsg/FormSG/pull/2131/commits/849a406d944d06ff91b945542351eaddbba31d30

**Review by karrui — APPROVED** [review]
lgtm

**Pull request merged**